pyresample/geometry.py

```diff
@@ -713,43 +715,301 @@ def copy(self):
+            extended_side2_lonlats = _get_extended_lonlats(*list_side2, npts=bottom)
+            dst_lats = np.concatenate((dst_lats, extended_side2_lonlats[1]), axis=0)
+            dst_lons = np.concatenate((dst_lons, extended_side2_lonlats[0]), axis=0)
+
+        if right > 0:
+            list_side1 = (dst_lons[:, -2], dst_lats[:, -2], dst_lons[:, -1], dst_lats[:, -1])
+            extended_side1_lonlats = _get_extended_lonlats(*list_side1, npts=right, transpose=False)
+            dst_lats = np.concatenate((dst_lats, extended_side1_lonlats[1]), axis=1)
+            dst_lons = np.concatenate((dst_lons, extended_side1_lonlats[0]), axis=1)
+
+        if left > 0:
+            list_side3 = (dst_lons[:, 1], dst_lats[:, 1], dst_lons[:, 0], dst_lats[:, 0])
+            extended_side3_lonlats = _get_extended_lonlats(*list_side3, npts=left, transpose=False)
+            dst_lats = np.concatenate((extended_side3_lonlats[1][:, ::-1], dst_lats), axis=1)
+            dst_lons = np.concatenate((extended_side3_lonlats[0][:, ::-1], dst_lons), axis=1)
+
+        # Back-conversion array as input format
+        lons, _ = _convert_2D_array(dst_lons, to=src_lons_format, dims=['y', 'x'])
+        lats, _ = _convert_2D_array(dst_lats, to=src_lats_format, dims=['y', 'x'])
+
+        # Add additional info if the source array is a DataArray
+        if isinstance(self.lats, xr.DataArray) and isinstance(self.lons, xr.DataArray):
+            lats.attrs = self.lats.attrs.copy()
+            lons.attrs = self.lons.attrs.copy()
+
+        # Return the extended SwathDefinition
+        return SwathDefinition(lons, lats)
+
+    def shrink(self, left=0, right=0, bottom=0, top=0):
+        """Shrink the SwathDefinition of n pixels on specific boundary sides.
+
+        By default, it does not shrink on any side.
+        """
+        # Check input validity
+        left = int(left)
+        right = int(right)
+        bottom = int(bottom)
+        top = int(top)
+        if left < 0 or right < 0 or bottom < 0 or top < 0:
+            raise ValueError('SwathDefinition.shrink expects positive numbers of pixels.')
+
+        # Return SwathDefinition if nothing to shrink
+        if left == 0 and right == 0 and bottom == 0 and top == 0:
+            return self
+
+        # Ensure shrinked area is at least 2x2
+        height = self.lats.shape[0]
+        width = self.lats.shape[1]
+        x_max_shrink = width - 2
+        y_max_shrink = height - 2
+        if (left + right) > x_max_shrink:
+            raise ValueError("SwathDefinition.shrink can drop maximum {} pixels "
+                             "along the x direction.".format(x_max_shrink))
+        if (top + bottom) > y_max_shrink:
+            raise ValueError("SwathDefinition.shrink can drop maximum {} pixels "
+                             "along the y direction.".format(y_max_shrink))
+
+        # Return the shrinked SwathDefinition
+        return self[slice(top, height - bottom), slice(left, width - right)]
```

Comment:
I think I mentioned this in a past comment, but why can't negative indexes be used here?

I don't see much utility here over just "normal" slicing. If `extend` has a real world use case I'd almost like to consider calling it something like `extrapolate` and getting rid of `shrink`.

Reply:
If `bottom=0` or `right=0`, `slice(top, -0)` or `slice(left, -0)` lead to empty arrays.

Reply:
Well dang. Ok...`None` might work but eh whatever.